### ISSUE TITLE
Scalping bot

### ISSUE THREAD

**Issue by mushbrian** [mushbrian/mushy#1]
import yfinance as yf
import pandas as pd
import sys
import pandas_datareader as pdr
import time
import requests
_zmq = DWX_ZeroMQ_Connector()

class buysold():
    #DWX_ZeroMQ_Connector()
    def buy():
        _my_trade = _zmq._generate_default_order_dict()
        _my_trade['_type'] = 0
        _my_trade['_symbol'] = 'EURUSDm'
        _my_trade['_price'] = 0.0
        _my_trade['_lots'] = 0.03
        _my_trade['_TP'] = 35
        _my_trade['_SL'] = 2000
        _my_trade[ '_magic'] = 123456
        _my_trade['_comment'] = 'mush,bought short' 
        _zmq._DWX_MTX_NEW_TRADE_(_order=_my_trade)

    def buy_now():
        _my_trade = _zmq._generate_default_order_dict()
        _my_trade['_type'] = 0
        _my_trade['_symbol'] = 'EURUSDm'
        _my_trade['_price'] = 0.0
        _my_trade['_lots'] = 0.03
        _my_trade['_TP'] =  70
        _my_trade['_SL'] = 2000
        _my_trade[ '_magic'] = 987654
        _my_trade['_comment'] = 'mush bought long'
        _zmq._DWX_MTX_NEW_TRADE_(_order=_my_trade)
        
    def sell():
        _my_trade = _zmq._generate_default_order_dict()
        _my_trade['_type'] = 1
        _my_trade['_symbol'] = 'EURUSDm'
        _my_trade['_price'] = 0.0
        _my_trade['_lots'] = 0.03
        _my_trade['_TP'] = 35
        _my_trade['_SL'] = 2000
        _my_trade[ '_magic'] = 234567
        _my_trade['_comment'] = ' mush,sold short'
        _zmq._DWX_MTX_NEW_TRADE_(_order=_my_trade)
        
    def sellnow():
        _my_trade = _zmq._generate_default_order_dict()
        _my_trade['_type'] = 1
        _my_trade['_symbol'] = 'EURUSDm'
        _my_trade['_price'] = 0.0
        _my_trade['_lots'] = 0.03
        _my_trade['_TP'] = 70
        _my_trade['_SL'] = 2000
        _my_trade[ '_magic'] = 876543
        _my_trade['_comment'] = ' long sell,mush'
        _zmq._DWX_MTX_NEW_TRADE_(_order=_my_trade)
    def  connected():
        url ="https://finance.yahoo.com/"
        try:
            requests.get(url, timeout= 5)
            return True
        except:
            requests.ConnectionError
            print(" connection lost , wait for internet")
            return False

   
    def  decision(window =12 , adjust = False):

       for i in range (7200):
            
            tick = "GC=F"
            e  =  yf.download(tick,period = "120h" ,interval = "5m")["Adj Close"]
            
            delta = e.diff(1).dropna()
            loss = delta.copy()
            gains= delta.copy()
            gains[gains<0]=0
            loss[loss>0] = 0
            gain_ewm = gains.ewm(com = window  ,adjust = adjust).mean()
            loss_ewm = abs(loss.ewm(com =window , adjust =adjust).mean())
            rs =gain_ewm/loss_ewm
            rsi = 100 - (100/(1+rs))
            s = rsi[-1]    
            #print (e)
            a= e[-1]
            b= e[-2]
            c= e[-5]
            d= e[-10]
            f= e[-15]
            g= e[-30]
            h= e[-45]
            k= e[-60]
            l = a+b+c+d+f+g+h+k+e[-20]+e[-40]
            #print(a)
            m = l / 10
            
            s3 = e.rolling(window =3).mean()[-1]
            s6 = e.rolling(window = 6).mean()[-1]
            s8 = e.rolling(window = 15).mean()[-1]
            s12= e.rolling(window =20).mean()[-1]
            sa  = (s6+s8+s12 +s3)/4
            w= (b +c+e[-3]+e[-4])/4# average of the last 5 closings 
            if  a > b and a >c  and a > e[-3 ] and s8 > s12   and a > s3 and s3> s6 and s6> s8 and  w>s8  and sa>s8   and s<64:
                 print("short buys")
                #buysold.buy()
                #buysold.buy()
            elif a > b and a > c and a > e[-3] and s8> s12 and a > s3 and s3 > s6 and s6 > s8 and w>s8 and sa>s8 and  s>71:
                print("short sells from   highhighs")
               # buysold.sell()
                #sell()
            elif a<b and a<c and a<e [-3] and a <e [-4] and a<s3 and s3<sa and s3 <w and s3<s6  and s6<s8  and w <s12 and  sa<s12 and s>29:
                print("short sells")
               # buysold.sell()
                #buysold.sell()
            elif a<b and a< c and a< e[-3] and a<e[-4 ] and a<s3 and s3<sa and s3<w and s3<s6 and s6<s8  and w<s12 and sa< s12  and s<24:
                print("short buys from lowlows")
              #  buysold.buy()
               # buysold.buy()
            elif a> b and b>c and c>d  and a>f  and a>g and a>h and a>k  and w >m and a>s3  and s3>s6 and s6>s8 and s8>s12 and a > w and s>71 and sa> s8:
                print("long sells from highhigher")
                #buysold.sellnow()
                #buysold.sellnow()
                #buysold.sellnow()  
            elif a<b and b<c and c<d  and a< f and  a< g and a < h  and a<k  and w<m  and a  < s3  and s6<s8 and sa <s8  and a< w  and s8 < s12 and s<30 and s3<s6:
                print("long buys from lowerlows")
              #  buysold.buy_now()
                #buysold.buy_now()
                #buysold.buy_now()   
            elif s3> s6 and s8> s12 and s6 >s8:
                _zmq._DWX_MTX_CLOSE_TRADES_BY_MAGIC_(234567)
                print("stop using magic number all sell trades except those sold from a highhigh")
            elif s3<s6 and s8<s12   and s6<s8:
                _zmq._DWX_MTX_CLOSE_TRADES_BY_MAGIC_(123456)
                print("stop all buy positions using magic number  except those bought from lower lows")
            else:
                print(a)
 
            time.sleep(30)
            i =+1

buysold.decision()
